var/spack/repos/builtin/packages/spectre/package.py

```diff
@@ -0,0 +1,189 @@
+    depends_on('gsl')
+    depends_on('hdf5@:1.12~mpi')
+    depends_on('jemalloc', when='memory_allocator=jemalloc')
+    depends_on('libsharp@1.0.0:1.0~mpi~openmp')
+    depends_on('libxsmm@1.16.1:1')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('yaml-cpp@0.6.3:0.7')
+
+    # Test dependencies
+    depends_on('catch2@2.8:2', type='test')
+    depends_on('py-numpy@1.10:1', type='test')
+    depends_on('py-scipy@1.0.0:1', type='test')
+    depends_on('py-h5py@3.0.0:3~mpi', type='test')
+
+    # Python bindings
+    with when('+python'):
+        extends('python')
+        depends_on('python@3.7:3.10', type=('build', 'run'))
+        depends_on('py-pybind11@2.6:2', type='build')
+        depends_on('py-numpy@1.10:1', type=('build', 'run'))
+        depends_on('py-scipy@1.0.0:1', type=('build', 'run'))
+        depends_on('py-matplotlib@3.0.0:3', type=('build', 'run'))
+        depends_on('py-h5py@3.0.0:3~mpi', type=('build', 'run'))
+
+    # Docs
+    with when('+doc'):
+        depends_on('doxygen@1.9.1:1.9', type='build')
+        depends_on('py-beautifulsoup4@4.0.0:4', type='build')
+        depends_on('py-pybtex@:0.24', type='build')
+
+    # TODO: Fix PCH builds
+    conflicts('+pch')
+
+    # Backport experimental support for Charm++ v7+
+    patch(
+        'https://github.com/sxs-collaboration/spectre/commit/a2203824ef38ec79a247703ae8cd215befffe391.patch',
+        sha256='eb6094028530d9f28cb9c91a90b4af908cc537c8525fb4c81b11c74fd0354932',
+        when='@:2022.01.03 ^charmpp@7.0.0:')
+    # Backport installation of targets, based on upstream patch:
+    # https://github.com/sxs-collaboration/spectre/commit/fe3514117c8205dbf18c4d42ec17712e67d33251
+    patch('install-pre-2022.01.03.patch', when='@:2022.01.03')
+    # Backport IWYU toggle to avoid CMake configuration issues
+    patch(
+        'https://github.com/sxs-collaboration/spectre/commit/cffeba1bc24bf7c00ec8bac710f02d3db36fa111.patch',
+        sha256='912877d4f553adff8b6df8264c50600c1e6d5a9c3ad18be0b86c9d801c07699c',
+        when='@:2022.01.03 ~pch')
+    # Backport patch for Boost 1.77
+    patch(
+        'https://github.com/sxs-collaboration/spectre/commit/001fc190a6ec73ad6c19ada9444d04a2320f2b96.patch',
+        sha256='bf539feb01d01e178889828dbbe5e990e8ee58c9e971d8634845c70a7cfb42a9',
+        when='@:2022.01.03 ^boost@1.77:')
+    # Backport patch for Python 3.10 in tests
+    patch(
+        'https://github.com/sxs-collaboration/spectre/commit/82ff2c39cdae0ecc1e42bdf4564506a4ca869818.patch',
+        sha256='5a5a3abf102e92812933e7318daabe2ca0a5a00d81d9663731c527e5dc6c8ced',
+        when='@:2022.01.03 ^python@3.10:')
+    # Backport `BUILD_TESTING` toggle
+    patch(
+        'https://github.com/sxs-collaboration/spectre/commit/79bed6cad6e95efadf48a5846f389e90801202d4.patch',
+        sha256='9e2a9bfadcd45ab154c151c0aa9af7c34ce0f2eb2f06f0eadc992d27568ce579',
+        when='@develop')
+    patch('build-testing-pre-2022.01.03.patch', when='@:2022.01.03')
```

Comment:
Are these patches scheduled to be released with SPeCTRE?

Reply:
These are updates to the build system that I made to support installation with Spack. They are all going to be included in the February release (the last one is still in a PR, https://github.com/sxs-collaboration/spectre/pull/3744). My intention here is to backport them to earlier releases, so they also install with Spack. In particular, I'd like to support release `2021.12.15` so the results of https://arxiv.org/abs/2108.05826 are reproducible.

Reply:
I would add a comment to each patch (or group of patches) stating this.